rebasebot/bot.py

```diff
@@ -163,6 +172,20 @@ def _do_rebase(gitwd, source, dest, source_repo, tag_policy):
             if not _resolve_rebase_conflicts(gitwd):
                 raise RepoException(f"Git rebase failed: {ex}") from ex
 
+    if update_go_modules:
+        logging.info(go_mod_commits)
+        # We have a list of all the bot commits with go mod updates
+        # We reset all these commits and delete them
+        # The flag `update_go_modules` will create new go mod updates commit
+        for commit in go_mod_commits:
+            try:
+                logging.info(f"cherry-picked this {commit}")
+                gitwd.git.cherry_pick(f"{commit}", "-Xtheirs")
+            except git.GitCommandError as ex:
+                if not _resolve_rebase_conflicts(gitwd):
+                    raise RepoException(f"Git rebase failed: {ex}") from ex
+        gitwd.git.reset("--hard", f"HEAD~{len(go_mod_commits)}")
```

Comment:
I don't think we actually need to do this, picking the commits and then doing a hard reset removes them again, so we don't need to pick them in the first place, I tried this branch without this section and I think it produces the same result, but with fewer commit conflict logs, so maybe it's better, can you try without this block and see if you agree?